File: LeetCode/C++/800-899/821.cpp
#include <iostream>
#include <vector>
#define MAXINT 99999
using namespace std;

class Solution {
public:
    vector<int> shortestToChar(string S, char C) {
        vector<int> ans(S.length());
        
        int prev = -MAXINT;
        for(int i = 0 ; i < S.length();++i){
            if(S[i] == C){
                prev = i;
            }
            
            ans[i] = i - prev;
        }
        
        prev = MAXINT;
        for(int i = S.length() - 1 ; i >= 0;--i){
            if(S[i] == C){
                prev = i;
            }
            
            ans[i] = min(ans[i], prev - i);
        }
        
        return ans;
    }
};

int main()
{
    Solution s;
	vector<int> ans = s.shortestToChar("loveleetcode", 'e');
	for(int num : ans){
		cout << num << endl;
	}

    return 0;
}

File: LeetCode/C#/800-899/821.cs
using System;

namespace LeetCode821
{
	public class Solution {
		private const int MAXINT = 99999;
		public int[] ShortestToChar(string S, char C) {
			int[] ans = new int[S.Length];
			
			int prev = -MAXINT;
			for(int i = 0 ; i < S.Length;++i){
				if(S[i] == C){
					prev = i;
				}
				
				ans[i] = i - prev;
			}
			
			prev = MAXINT;
			for(int i = S.Length - 1 ; i >= 0;--i){
				if(S[i] == C){
					prev = i;
				}
				
				ans[i] = Math.Min(ans[i], prev - i);
			}
			
			return ans;
		}
	}
	
	public class Program
	{
		public static void Main()
		{
			Solution sol = new Solution();
			int[] ans = sol.ShortestToChar("loveleetcode" , 'e');
			foreach(int num in ans)
			{
				Console.WriteLine(num);
			}

			Console.Read();
		}
	}
}